### PR TITLE
Add Git Hooks to run lint

### PR DESCRIPTION
**Pull request by PikachuEXE**
**Pull Request Type**
Please select what type of pull request this is:
- [x] Feature Implementation

**Related issue**
Smaller version of #1171

**Description**
For new comers not coming from #1171 
Git Hooks is `Git has a way to fire off custom scripts when certain important actions occur.` (quote from official doc)
There are client-side and server-side hooks but this PR is only about the client-side hooks.

For all available hook events and other stuff about Git Hooks see:
https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks

**Summary of changes**
- Add git hook via `lefthook`
- Git Hook scripts are generated by `lefthook` when installed via `yarn`, no extra step required
- Add a `pre-commit` hook to run `eslint` on every commit which lints staged (about to be committed) files only (faster) where the filenames end with `js/ts/vue`

`lefthook` is picked as git hook manager due to
- Written in Go
  - `Go language makes Lefthook lightning-fast and provides support for concurrently executed scripts out of the box.`
  - It can be installed without too many dependencies on many environments via different methods (But since we use `yarn`, just use `yarn`)
- Run multiple commands in one hook, in sequence or in parallel (`simple-git-hooks` can't run commands in parallel)
- Installing `lefthook` via `npm`/`yarn` will generate git hook scripts in your git hook path
  - No extra setup step = no chance for forgetting to setup git hook

Read [the introduction post for `lefthook`](https://evilmartians.com/chronicles/lefthook-knock-your-teams-code-back-into-shape) if you interested in finding out more

**PR Purpose**
Lint task is sometimes NOT run by the PR submitter (including me!)
I know we have server side GH Action checks to ensure
So this is just an dev process improvement which should add just a little time to commit changes

Also #1171 is too big to review and merge

**Screenshots (if appropriate)**
Code Change Diff (for testing linting)
Screenshot 2021-10-04 at 3 27 02 PM

Git output
image

**Testing (for code that is not small enough to be easily understandable)**
- Make change(s) violating any lint rule
- Commit those changes

**Desktop (please complete the following information):**
 - OS: MacOS
 - OS Version: 11.6
 - FreeTube version: based on 84ac98f0a33c8ca450019c6ea7d43abd02c2fb2c

**Additional context**

### CODE REVIEW

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]
LGTM

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]

**Review by absidue — CHANGES_REQUESTED** [review]

File: lefthook.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Refer for explanation to following link:
+# https://github.com/evilmartians/lefthook/blob/master/docs/full_guide.md
+pre-commit:
+  parallel: true
+  commands:
+    lint:
+      # Only runs when any file with filename
+      # matching the glob is being committed
+      glob: "*.{js,ts,vue}"
+      run: npx eslint --no-color {staged_files}
```
[the remark this answers is not in the document]
Comment (PikachuEXE) [reply lefthook.yml]: (Unrelated to this PR)
Do you think we should also replace other `npm run` in CI to yarn?

**Comment by PikachuEXE** [comment]
All requested changes in https://github.com/FreeTubeApp/FreeTube/pull/1788#pullrequestreview-1016722307 made

**Review by absidue — APPROVED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Comment by PikachuEXE** [comment]
@ChunkyProgrammer ~

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@ChunkyProgrammer one more approval needed. Review if u have some spare time please

**Comment by ChunkyProgrammer** [comment]
No idea how I missed the notifications for this one. This looks good to me. I'll approve it now. (There are newer versions of lefthook available so maybe it should be upgraded soon?)

**Review by ChunkyProgrammer — APPROVED** [review]

**Comment by PikachuEXE** [comment]
Crap I forgot to update the lefthook version

**Pull request merged**